src/Accelerators/NNPA/Runtime/zDNNExtension/zDNNExtension.c

```diff
@@ -103,42 +103,145 @@ static zdnn_status allocZTensorChunk(
+      preTransDesc->dim2 = chunkSize;
+    if (axis == 3)
+      preTransDesc->dim1 = chunkSize;
+    break;
+  case (ZDNN_4D):
+  case (ZDNN_NHWC):
+  case (ZDNN_HWCK):
+    // shape (a, b, c, d) -> dims4-1 (a, b, c, d)
+    // shape (n, h, w, c) -> dims4-1 (n, h, w, c)
+    if (axis == 0)
+      preTransDesc->dim4 = chunkSize;
+    if (axis == 1)
+      preTransDesc->dim3 = chunkSize;
+    if (axis == 2)
+      preTransDesc->dim2 = chunkSize;
+    if (axis == 3)
+      preTransDesc->dim1 = chunkSize;
+    break;
+  case (ZDNN_4DS):
+    // ZDNN_4DS is used exclusively as RNN output
+    // shape (a, b, c, d)  -> ZDNN_NHWC
+    //   when b = 1 (uni-dir)     -> dims4-1 (a, 1, c, d)
+    //   otherwise (bi-dir, etc.) -> dims4-1 (a, 1, c, b * PADDED(d))
+    if (axis == 0)
+      preTransDesc->dim4 = chunkSize;
+    if (axis == 2)
+      preTransDesc->dim2 = chunkSize;
+    if (axis == 3)
+      preTransDesc->dim1 = chunkSize;
+    break;
+  case (ZDNN_NCHW):
+    // shape (n, c, h, w) -> dims4-1 (n, h, w, c)
+    if (axis == 0)
+      preTransDesc->dim4 = chunkSize;
+    if (axis == 1)
+      preTransDesc->dim2 = chunkSize;
+    if (axis == 2)
+      preTransDesc->dim1 = chunkSize;
+    if (axis == 3)
+      preTransDesc->dim3 = chunkSize;
+    break;
+  default:
+    break;
+  }
 
   // Copy a transformed desc.
   zdnn_status status = zdnn_generate_transformed_desc(preTransDesc, transDesc);
   if (status != ZDNN_OK)
     return status;
 
-  // Init a zTensor with malloc.
-  return zdnn_init_ztensor_with_malloc(preTransDesc, transDesc, chunkZTensor);
+  if (splitInfo->reuseOrigBuffer) {
+    zdnn_init_ztensor(preTransDesc, transDesc, chunkZTensor);
+    // Set a buffer size for the chunk.
+    chunkZTensor->buffer_size = zdnn_getsize_ztensor(transDesc);
+    // Set a buffer for the chunk.
+    uint64_t reuseBufferOffset;
+    if (chunkID == splitInfo->numOfChunks - 1) {
+      assert((origZTensor->buffer_size - chunkZTensor->buffer_size) %
+                 (splitInfo->numOfChunks - 1) ==
+             0);
+      reuseBufferOffset =
+          (origZTensor->buffer_size - chunkZTensor->buffer_size) /
+          (splitInfo->numOfChunks - 1) * chunkID;
+      // Make sure offset is 4K-alignment.
+      assert((reuseBufferOffset % AIU_PAGESIZE_IN_BYTES == 0) &&
+             "Buffer is not 4K-alignment");
+    } else {
+      // This is always 4K-alignment.
+      reuseBufferOffset = chunkZTensor->buffer_size * chunkID;
+    }
+    chunkZTensor->buffer = origZTensor->buffer + reuseBufferOffset;
+    // Make sure the chunk buffer is within the original buffer.
+    assert(((reuseBufferOffset + chunkZTensor->buffer_size) <=
+               origZTensor->buffer_size) &&
+           "Chunk buffer is outside the original buffer");
+    status = ZDNN_OK;
+  } else {
+    // Init a zTensor with malloc.
+    status =
+        zdnn_init_ztensor_with_malloc(preTransDesc, transDesc, chunkZTensor);
+  }
+  chunkZTensor->is_transformed = true;
+
+  return status;
```

Comment:
It would be useful for parallelization if you could state at the beginning of this function that it only modifies values inside the chunkInfo data structure `splitInfo.chunk[chunkID]`, if true.